src/object/obj_internal.h

```diff
@@ -80,30 +75,37 @@ struct dc_obj_shard {
 #define do_fseq		do_pl_shard.po_fseq
 #define do_rebuilding	do_pl_shard.po_rebuilding
 
+/** client object shards head */
+struct dc_obj_shards_head {
```

Comment:
If we are adding this data structure, do we still want to have dc_obj_shard::do_obj, or we can put it in this data structure so we can save a few bytes.
Also, I would just call it dc_obj_layout

Reply:
The dc_obj_shard::do_obj is used as the clew for whether the shard is opened or not, such as obj_layout_free(). It is not impossible to replace it with other an int flag, but it does not bring too much benefit if consider alignment.